tree/ntuple/v7/test/ntuple_storage_daos.cxx

```diff
@@ -3,23 +3,55 @@
 #include "ROOT/TestSupport.hxx"
 #include <iostream>
 
-TEST(RPageStorageDaos, Basics)
-{
-   ROOT::TestSupport::CheckDiagsRAII diags;
-   diags.optionalDiag(kWarning, "in int daos_init()", "This RNTuple build uses libdaos_mock. Use only for testing!");
-   diags.requiredDiag(kWarning, "ROOT::Experimental::Detail::RPageSinkDaos::RPageSinkDaos",
-                      "The DAOS backend is experimental and still under development.", false);
-   diags.requiredDiag(kWarning, "[ROOT.NTuple]", "Pre-release format version: RC 1", false);
+class RPageStorageDaos : public ::testing::Test {
+public:
+   std::unique_ptr<ROOT::TestSupport::CheckDiagsRAII> fTestRootDiags;
+
+   static constexpr const char fDaosContainerLabels[5][22] = {
+      "test-basics", "test-extended", "test-options", "test-multiple-ntuples", "test-caged-pages",
+   };
+
+   [[nodiscard]] static std::string GetDaosUri(int testIndex)
```

Comment:
I find it much simpler to directly pass a `std::string_view` as an argument to the `SetUp()` function (see https://stackoverflow.com/a/38218657).  Therefore, we should be able to call
```c++
SetUp("test-container-xxx");
```
at the beginning of each test.

Provided that the `SetUp()` function registers the container label in a `static` member of type `std::unordered_set<std::string>`, the following should be also possible:
- `GetDaosUri()` -no arguments- should be trivial to implement.
- `TearDownTestSuite()` should be able to iterate over all the previously registered containers in all the tests

IMHO, the suggested changes help readability while making it more straightforward to add new tests.

Reply:
That does seem much more readable and extensible. But what about rolling `GetDaosUri(void)` into the new `SetUp(string_view)` function (potentially renamed) and let that return the URI, forgoing the addl `string` attribute? See the next commit(s)!